include/cuco/detail/trie/bit_vector/bit_vector_ref.inl

```diff
@@ -0,0 +1,238 @@
+#include <cuco/detail/trie/bit_vector/bit_vector.cuh>
+
+namespace cuco {
+namespace experimental {
+
+template <typename StorageRef, typename... Operators>
+__host__ __device__ constexpr bit_vector_ref<StorageRef, Operators...>::bit_vector_ref(
+  StorageRef words_ref,
+  StorageRef ranks_ref,
+  StorageRef selects_ref,
+  StorageRef ranks0_ref,
+  StorageRef selects0_ref) noexcept
+  : words_ref_{words_ref},
+    ranks_ref_{ranks_ref},
+    selects_ref_{selects_ref},
+    ranks0_ref_{ranks0_ref},
+    selects0_ref_{selects0_ref}
+{
+}
+
+namespace detail {
+
+template <typename StorageRef, typename... Operators>
+class operator_impl<op::bv_read_tag, bit_vector_ref<StorageRef, Operators...>> {
+  using ref_type  = bit_vector_ref<StorageRef, Operators...>;  ///< Bitvector ref type
+  using size_type = typename StorageRef::size_type;            ///< Size type
+  using slot_type = typename StorageRef::value_type;           ///< Slot type
+
+  const size_type bits_per_word   = sizeof(slot_type) * 8;
+  const size_type words_per_block = 4;  //< This should match the defintion in bit_vector
```

Comment:
These can be `constexpr`.

Reply:
9d5100d2c20347acf18d4d10a30ea76db159f13e